core/model/src/main/java/org/eclipse/rdf4j/model/util/Configurations.java

```diff
@@ -47,17 +50,17 @@ public class Configurations {
 	@InternalUseOnly
 	public static Optional<Resource> getResourceValue(Model model, Resource subject, IRI property,
 			IRI legacyProperty) {
-		var result = Models.objectResource(model.getStatements(subject, property, null));
+		var result = Models.objectResource(model.getStatements(subject, legacyProperty, null));
 		if (result.isPresent()) {
 			return result;
 		}
-		return Models.objectResource(model.getStatements(subject, legacyProperty, null));
+		return Models.objectResource(model.getStatements(subject, property, null));
```

Comment:
Why do you prefer prioritising legacy predicates?

Reply:
Users who modify the model would be using the current (legacy) vocabulary and if there is any discrepancy between what the new vocabulary says and the current (legacy) vocabulary then we should pick the current (legacy) vocabulary.

A very simple example:
```
  [] legacy:enabled true;
     new:enabled false.
```